pre_commit_hooks/auto_indent.py

```diff
@@ -0,0 +1,183 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from __future__ import print_function
+
+import argparse
+import ast
+import io
+import re
+import textwrap
+import traceback
+from operator import itemgetter
+
+
+RETURN_CODE = dict(
+    no_change=0,
+    fixed_file=1,
+    syntax_error=2,
+)
+
+
+class IndentNodeVisitor(ast.NodeVisitor):
+
+    def __init__(self):
+        self.lines = []
+
+    def visit_Call(self, node):
+        args = list(node.args)
+
+        if node.keywords:
+            args.extend([keyword.value for keyword in node.keywords])
+
+        if node.starargs:
+            args.append(node.starargs)
+
+        if node.kwargs:
+            args.append(node.kwargs)
+
+        try:
+            if args and args[0].lineno == node.lineno:
+                for arg in args:
+                    if arg.lineno != node.lineno:
+                        self.lines.append({
+                            'line': node.lineno,
+                            'nextline': arg.lineno,
+                            'column': args[0].col_offset,
+                        })
+                        break
+        except AttributeError:
+            # Python 3.3 is not supported because line numbers were
+            # removed from arg objects (compared to Python 2.7 and Python 3.4):
+            # https://docs.python.org/3.3/library/ast.html
+            pass
+        finally:
+            self.generic_visit(node)
+
+    def visit_ClassDef(self, node):
+        if node.bases and node.bases[0].lineno != node.lineno:
+            self.generic_visit(node)
+            return
+
+        for base in node.bases:
+            if base.lineno != node.lineno:
+                self.lines.append({
+                    'line': node.lineno,
+                    'nextline': base.lineno,
+                    'column': node.bases[0].col_offset,
+                })
+
+                break
+
+        self.generic_visit(node)
+
+    def visit_FunctionDef(self, node):
+        try:
+            if node.args.args and node.args.args[0].lineno != node.lineno:
+                self.generic_visit(node)
+                return
+
+            for arg in node.args.args:
+                if arg.lineno != node.lineno:
+                    self.lines.append({
+                        'line': node.lineno,
+                        'nextline': arg.lineno,
+                        'column': node.args.args[0].col_offset,
+                    })
+
+                    break
+        except AttributeError:
+            # Python 3.3 is not supported because line numbers were
+            # removed from arg objects (compared to Python 2.7 and Python 3.4):
+            # https://docs.python.org/3.3/library/ast.html
+            pass
+        finally:
+            self.generic_visit(node)
+
+
+def check_files(files):
+    rc = RETURN_CODE['no_change']
+
+    for fpath in files:
+        with open(fpath) as f:
```

Comment:
use io.open with an explicit open mode -- right now you're opening as binary in python2 and text in python3